api/v1alpha1/template_types.go

```diff
@@ -37,28 +39,32 @@ const (
 	ChartAnnotationControlPlaneProviders = "hmc.mirantis.com/control-plane-providers"
 )
 
-// TemplateType specifies the type of template packaged as a helm chart.
-// Should be provided in the chart Annotations.
-type TemplateType string
+type TemplateKind string
 
-const (
-	// TemplateTypeDeployment is the type used for creating HMC ManagedCluster objects
-	TemplateTypeDeployment TemplateType = "deployment"
-	// TemplateTypeProvider is the type used for adding CAPI providers in the HMC Management object.
-	TemplateTypeProvider TemplateType = "provider"
-	// TemplateTypeCore is the type used for HMC and CAPI core components
-	TemplateTypeCore TemplateType = "core"
-)
+// +kubebuilder:object:generate=false
+// +k8s:deepcopy-gen=false
+
+// Template is the interface defining a list of methods to interact with templates
+type Template interface {
+	Get() TemplateState
+	UpdateStatus(context.Context, crclient.Client, TemplateStatusMixin) error
+}
+
+// +kubebuilder:object:generate=false
+// +k8s:deepcopy-gen=false
 
-// TemplateSpec defines the desired state of Template
-type TemplateSpec struct {
+type TemplateState struct {
```

Comment:
I don't think we need a dedicated struct. Let's add getters that return pointers to status/spec instead.